src/mobile-ui-react/MobileUi.tsx

```diff
@@ -558,11 +582,10 @@ export function useIsolatedCount(): number {
  * @param handler - The callback function.
  */
 export function useIModel(handler: (iModel: IModelConnection | undefined) => void) {
-  useSyncUiEvent(React.useCallback(() => {
-    handler(UiFramework.getIModelConnection());
-    // @todo AppUI deprecation
-    // eslint-disable-next-line deprecation/deprecation
-  }, [handler]), SessionStateActionId.SetIModelConnection);
+  const iModelConnection = useActiveIModelConnection();
```

Comment:
I would suggest removing this function since it doesn't add anything that can't be accomplished with `useActiveIModelConnection`

Reply:
SYNCHRO Field uses it in one place, and they take advantage of the fact that they get a callback when it happens. They don't actually care about the value, simply that it changed:

```ts
  useIModel(() => {
    reset();
  });
```

Obviously they could update their code to `React.useMemo` like I do here. I guess we could ask them.

Reply:
I would suggest calling `useEffect` instead of `useMemo` since the later is meant for memoizing calculated data (i.e. storing the return value of `useMemo`).

I think having Field update their code to call `useActiveIModelConnection` makes sense.

Reply:
If SYNCHRO agrees with us removing useIModel, this discussion is moot, but based on my understanding of how useEffect and useMemo work, useEffect will cause an extra render at startup and every time the value changes. Having said that, I am well aware that this is a very non-standard usage of useMemo, so maybe the extra render is not something to worry about.